qa/rpc-tests/liststucktransactions.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Dogecoin Core Developers
+# Distributed under the MIT software license, see the accompanying
+# file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+"""Stuck Transaction Test
+
+# Tests liststucktransactions RPC endpoint
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+from test_framework.mininode import wait_until
+from decimal import Decimal
+
+class ListStuckTransactionsTest(BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+        self.mocktime = int(time.time())
+        self.burn_addr = "n4LRQGEKcyRCXqD2MH3ompyMTJKitxu1WP" # no one has this
+
+    def setup_nodes(self, split=False):
+        nodes = []
+
+        # Our wallet
+        nodes.append(start_node(0, self.options.tmpdir,
+            ["-acceptnonstdtxn=0", "-discardthreshold=0.1", "-mempoolexpiry=1", "-paytxfee=0.1"]))
+
+        # A miner that doesn't allow tx under 1 DOGE
+        nodes.append(start_node(1, self.options.tmpdir,
+            ["-acceptnonstdtxn=0", "-softdustlimit=1", "-harddustlimit=1", "-mempoolexpiry=1", "-debug=net"]))
+
+        return nodes
+
+    def setup_network(self, split = False):
+        self.nodes = self.setup_nodes()
+
+        # set the time explicitly
+        self.setmocktime(int(time.time()))
```

Comment:
Is this redundant with the assignment in the initializer?

I don't feel strongly about this.

Reply:
I played with the thought of removing the initializer assignment (this also sets it on the nodes, so this one has to stay) but I thought it safer to initialize to a meaningful value there. May be overly defensive though... 

We can call `incrementmocktime(1)` or something but I felt that would be less clear than just explicitly setting the current time (which is on most platforms a second or 2 later than when we did init, because of the time it takes to start up nodes and verify them)

Reply:
That makes sense. I'm fine with this as it is then.